### ISSUE TITLE
Fody Add-In testing with proper linking to modified assemblies

### ISSUE THREAD

**Issue by jtheisen** [Fody/Fody#356]
I wrote a Fody Add-In (similar to `PropertyChanged`) and now it's time to write some tests.

I don't like how the `BasicFodyAddin` and `PropertyChanged` does all their testing using `dynamic` to avoid linking against the libraries.

On the other hand, when one does link against the libraries the build process has to account for a weaving before the tests run (as it's no longer the tests themselves doing the weaving).

Is there another Add-In I could use as a template for this sort of thing or am I the only one who would prefer it that way?

**Comment by distantcam** [comment]
Pretty sure you're the only one who prefers this way. ;)

If you want to avoid dynamic linking you can have an interface that implements all the methods to test, and then link that to the tests so the assembly to test isn't directly linked, but you don't have to use `dynamic`.

**Comment by jtheisen** [comment]
What I now ended up doing in `Loom.Fody` is that `AssemblyToModify` has the `Fody` package and is thus modified in the same way as when you use the add-in during build. I had to tweak the paths a little and I had to have a look at Fody's build tasks to see where it's really looking, but it does work now.

That makes testing the weaving awkward, but I rarely had anything to debug there anyway. And it makes testing the woven end-product easier.
